spec/lib/composer_messages_finder_spec.rb

```diff
@@ -328,6 +328,68 @@
     end
   end
 
+  describe ".dont_feed_the_trolls" do
+    fab!(:user) { Fabricate(:user) }
+    fab!(:author) { Fabricate(:user) }
+    fab!(:other_user) { Fabricate(:user) }
+    fab!(:third_user) { Fabricate(:user) }
+    fab!(:topic) { Fabricate(:topic, user: author) }
+    fab!(:unflagged_post) { Fabricate(:post, topic_id: topic.id, user: author) }
+    fab!(:self_flagged_post) { Fabricate(:post, topic_id: topic.id, user: author) }
+    fab!(:under_flagged_post) { Fabricate(:post, topic_id: topic.id, user: author) }
+    fab!(:over_flagged_post) { Fabricate(:post, topic_id: topic.id, user: author) }
+
+    before { SiteSetting.dont_feed_the_trolls_threshold = 2 }
+
+    it "does not show a message for unflagged posts" do
+      finder =
+        ComposerMessagesFinder.new(
+          user,
+          composer_action: "reply",
+          topic_id: topic.id,
+          post_id: unflagged_post.id,
+        )
+      expect(finder.check_dont_feed_the_trolls).to be_blank
+    end
+
+    it "shows a message when the replier has already flagged the post" do
+      Fabricate(:flag, post: self_flagged_post, user: user)
+      finder =
+        ComposerMessagesFinder.new(
+          user,
+          composer_action: "reply",
+          topic_id: topic.id,
+          post_id: self_flagged_post.id,
+        )
+      expect(finder.check_dont_feed_the_trolls).to be_present
+    end
+
+    it "does not show a message when not enough others have flagged the post" do
+      Fabricate(:flag, post: under_flagged_post, user: other_user)
+      finder =
+        ComposerMessagesFinder.new(
+          user,
+          composer_action: "reply",
+          topic_id: topic.id,
+          post_id: under_flagged_post.id,
+        )
+      expect(finder.check_dont_feed_the_trolls).to be_blank
+    end
+
+    it "shows a message when enough others have already flagged the post" do
+      Fabricate(:flag, post: over_flagged_post, user: other_user)
+      Fabricate(:flag, post: over_flagged_post, user: third_user)
+      finder =
+        ComposerMessagesFinder.new(
+          user,
+          composer_action: "reply",
+          topic_id: topic.id,
+          post_id: over_flagged_post.id,
+        )
+      expect(finder.check_dont_feed_the_trolls).to be_present
```

Comment:
I think we're missing a system test here because we're not asserting for what is being returned by the method. That means this feature could potentially break without us knowing about it should the details of the hash that is being returned by the method change.